packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -286,10 +286,9 @@ class ImageCache {
     }
   }
 
-  void _trackLiveImage(Object key, _LiveImage image, { bool debugPutOk = true }) {
+  void _trackLiveImage(Object key, _LiveImage image) {
```

Comment:
For future us:

This check was bogus. The assumption was that if we were trying to track a live image from the listener below and we didn't already know about it, we had either (a) missed a branch where we should have tracked it previously or (b) were being called synchronously, and thus hadn't had a chance to hit all the branches.

The problem is that even if we're not called synchronously, someone could have cleared the live image (e.g. by calling `clearLiveImages`) before the code reached this point asynchronously, which would then trigger the assert. While it's very strange to do that, we shouldn't bother with asserting about it, and in fact it was happening in the test(s) that were failing here.